src/pages/Home/Home.tsx

```diff
@@ -25,70 +26,57 @@ export default function Home() {
+        if (scrollPosition >= category.offsetY - HEADER_HEIGHT) {
+          activeIndex++;
+        }
+      });
+      setActiveCategory(categories[activeIndex].name);
     }
-    function loadHandler() {
-      setLoaded(true);
-    }
-    window.addEventListener("scroll", scrollHandler);
-    window.addEventListener("load", loadHandler);
-
-    let categoryAndPosition: CategoryPosition[] = [];
-
-    const SectionElements = contentDivRef.current?.getElementsByClassName(
-      "FoodCategorySectionHeader"
-    );
-    const length = SectionElements?.length ? SectionElements.length : 0;
-
-    //BUGFIX
-    if (!loaded) {
-      console.log("not loaded!");
-    } else {
-      console.log("loaded");
-    }
-
-    for (let i = 0; i < length; i++) {
-      let offsetY =
-        (SectionElements?.item(i)?.getBoundingClientRect().top as number) +
-        window.scrollY;
-      if (i !== 0) offsetY += BORDER_TOP_WIDTH;
-      const categoryName = SectionElements?.item(i)?.firstChild
-        ?.textContent as string;
-      categoryAndPosition.push({ name: categoryName, offsetY: offsetY });
-    }
-    console.log(categoryAndPosition);
-    setCategories(categoryAndPosition);
-    setActiveCategory(categoryAndPosition[0].name);
 
+    window.addEventListener("scroll", scrollHandler);
     return () => {
       window.removeEventListener("scroll", scrollHandler);
-      window.removeEventListener("load", loadHandler);
     };
-  }, [loaded]);
+  }, [categories]);
 
-  //on scroll, check for change in activeCategory
   useEffect(() => {
-    if (categories.length === 0) return; //type guard against null;
-
-    let activeIndex = -1;
-
-    //binary search would be better
-    categories.forEach((category) => {
-      if (scrollPosition >= category.offsetY - HEADER_HEIGHT) {
-        activeIndex++;
-      } else {
-        return;
+    // console.log("Mounted");
+    function loadHandler() {
+      // console.log("Loaded");
+      let categoryAndPosition: CategoryPosition[] = [];
+
+      const SectionElements = contentDivRef.current?.getElementsByClassName(
+        "FoodCategorySectionHeader"
+      );
+      const length = SectionElements?.length ? SectionElements.length : 0;
+
+      for (let i = 0; i < length; i++) {
+        let offsetY =
+          (SectionElements?.item(i)?.getBoundingClientRect().top as number) +
+          window.scrollY;
+        if (i !== 0) offsetY += BORDER_TOP_WIDTH;
+        const categoryName = SectionElements?.item(i)?.firstChild
+          ?.textContent as string;
+        categoryAndPosition.push({ name: categoryName, offsetY: offsetY });
       }
-    });
-    setActiveCategory(categories[activeIndex].name);
-  }, [scrollPosition, categories]);
+      setCategories(categoryAndPosition);
+      setActiveCategory(categoryAndPosition[0].name);
+    }
+    loadHandler();
+
+    window.addEventListener("load", loadHandler);
+    return () => {
+      window.removeEventListener("load", loadHandler);
+    };
+  }, []);
```

Comment:
`loadHandler()` has to be called on both onLoad and onMount, using the `loadEventListener`, and `useEffect `with no dependencies.

Issue when navigating back to home page after clicking on a food item, there won't be a loadEvent. Home page already loaded, even though we navigate back.
This means that the load handler won't run when navigating back  to home page. Categories won't be initialised. When navigating back, Home is mounted, but not loaded. So loadHandler has to be called both **onLoad** and **onMount** to run correctly.